airflow/www/views.py

```diff
@@ -1365,8 +1450,12 @@ def clear(self):
                                    recursive=recursive, confirmed=confirmed, only_failed=only_failed)
 
     @expose('/dagrun_clear', methods=['POST'])
-    @has_dag_access(can_dag_edit=True)
-    @has_access
+    @auth.has_access([
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAGS),
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK),
+        (permissions.ACTION_CAN_DELETE, permissions.RESOURCE_TASK_INSTANCE),
```

Comment:
```suggestion
        (permissions.ACTION_CAN_DELETE, permissions.RESOURCE_DAG_RUN),
```

Possibly _also_ can_delete TI